x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/recall_from_search_connectors.ts

```diff
@@ -155,19 +155,18 @@ async function recallFromLegacyConnectors({
         bool: {
           should: [
             {
-              text_expansion: {
-                [vectorField]: {
-                  model_text: text,
-                  model_id: modelId,
-                  boost,
-                },
+              sparse_vector: {
+                field: vectorField,
+                query: text,
+                inference_id: AI_ASSISTANT_KB_INFERENCE_ID,
```

Comment:
This will re-use the inference endpoint used for the internal KB. For now this is using elser2 but that is configurable, meaning a user could change it to e5. 
This is currently not a (big) issue since we haven't exposed it but we should probably have a separate inference endpoint that is guaranteed to point to Elser. Most likely we should use the preconfigured elser endpoint `.elser-2-elasticsearch`.